File: ios/HybridGoogleMapsNitroModule.swift
final class HybridGoogleMapsNitroModule: HybridGoogleMapsNitroModuleSpec {
  private let permissionHandler: PermissionHandler
  private let locationHandler: LocationHandler

  override init() {
    self.permissionHandler = PermissionHandler()
    self.locationHandler = LocationHandler()
  }

  func showLocationDialog() {
    locationHandler.showLocationDialog()
  }

  func openLocationSettings() {
    locationHandler.openLocationSettings()
  }

  func requestLocationPermission()
    -> NitroModules.Promise<RNLocationPermissionResult> {
    return permissionHandler.requestLocationPermission()
  }

  func isGooglePlayServicesAvailable() -> Bool {
    /// not supported
    return true
  }
}

File: android/src/main/java/com/googlemapsnitro/HybridGoogleMapsNitroModule.kt
package com.googlemapsnitro

import com.margelo.nitro.core.Promise

class HybridGoogleMapsNitroModule : HybridGoogleMapsNitroModuleSpec() {
  val context = GoogleMapsNitroPackage.AppContextHolder.context
  private val locationHandler: LocationHandler = LocationHandler(context)
  private val permissionHandler: PermissionHandler = PermissionHandler(context)
  private val playServicesHandler: PlayServicesHandler = PlayServicesHandler(context)

  override fun showLocationDialog() {
    locationHandler.showLocationDialog()
  }

  override fun openLocationSettings() {
    locationHandler.openLocationSettings()
  }

  override fun requestLocationPermission(): Promise<RNLocationPermissionResult> = permissionHandler.requestLocationPermission()

  override fun isGooglePlayServicesAvailable(): Boolean = playServicesHandler.isPlayServicesAvailable()
}
